src/binary/Yast.cc

```diff
@@ -166,6 +168,81 @@ void set_ruby_source_location(VALUE file, VALUE lineno)
   YaST::ee.setLinenumber(FIX2INT(lineno));
 }
 
+/**
+ * Is the function name an UI function which returns a symbol?
+ * @param  function_name name of the function
+ * @return true if function_name is one of "UserInput", "TimeoutUserInput"
+ *   or "PollInput"
+ */
+static bool ui_symbol_function(const char *function_name)
+{
+    return strcmp(function_name, "UserInput") == 0  ||
+        strcmp(function_name, "TimeoutUserInput") == 0 ||
+        strcmp(function_name, "PollInput") == 0;
+}
+
+/**
+ * Is the YCP value :debugHotkey symbol?
+ * @param  val YCP value
+ * @return true when val == :debugHotkey
+ */
+static bool is_debug_symbol(YCPValue val)
+{
+    return !val.isNull() && val->isSymbol() &&
+        val->asSymbol()->symbol() == "debugHotkey";
+}
+
+/**
+ * Is the function name an UI function which returns a map?
+ * @param  function_name name of the function
+ * @return true if function_name is "WaitForEvent"
+ */
+static bool ui_hash_function(const char *function_name)
+{
+    return strcmp(function_name, "WaitForEvent") == 0;
+}
+
+/**
+ * Is the YCP value a map containg a debug UI event?
+ * @param  val YCP value
```

Comment:
quite useless comment. In fact it expect result of UI event. So something like `@param  val YCPMap response from UI Event call` ? or at least UI event map